### ISSUE TITLE
`check-script-has-no-table-name`: double-quoted string literals not stripped, causing false positives when they contain the word FROM

### ISSUE THREAD

**Issue by roykoand** [dbt-checkpoint/dbt-checkpoint#379]
Bug:
```sql
select
    case
        when status = 'active' then "Upgraded From Plan A"
        else "Select Option From List"
    end as label
from {{ ref('my_model') }}
```
```python
python3 -c "
from dbt_checkpoint.check_script_has_no_table_name import has_table_name
print(has_table_name(open('repro.sql').read(), 'repro.sql'))
"
```

Expected output: `(0, set())`
Actual output: `(1, {'list"', 'plan'})`
